src/secure-value.ts

```diff
@@ -102,16 +101,32 @@ export const serializedSecureEnvSourceText = (function secureEnvFactory(rawEnv:
         return obj === undefined;
     }
 
+    function isNull(obj: any): obj is null {
+        return obj === null;
+    }
+
     function isFunction(obj: any): obj is Function {
         return typeof obj === 'function';
     }
 
-    function isNullish(obj: any): obj is null {
-        // eslint-disable-next-line eqeqeq
-        return obj == null;
+    function isNullOrUndefined(obj: any): obj is (null | undefined) {
+        return isNull(obj) || isUndefined(obj);
     }
 
     function getSecureValue(raw: RawValue): SecureValue {
+        if (isNullOrUndefined(raw)) {
+            return raw as SecureValue;
+        }
+        const t = typeof raw;
+        // NOTE: internationally checking for typeof 'undefined' for the case of
+        // `typeof document.all === 'undefined'`, which is an exotic object with
+        // a bizarre behavior described here:
+        // * https://tc39.es/ecma262/#sec-IsHTMLDDA-internal-slot
+        // This check cover that case, but doesn't affect other unidefined values
+        // because those are covered by the previous condition anyways.
+        if (t === 'function' || t === 'undefined') {
```

Comment:
this is the condition that includes `typeof document.all === 'undefined'` considering that if the value is truly `undefined`, it will exit in the block above not here.